docs/dev-guide/contributing/branching.rst

```diff
@@ -33,6 +33,18 @@ Each ``x.y`` release will have one corresponding branch called ``x.y-dev``. For
 example, all work for the 2.7.z series of releases gets merged into ``2.7-dev``.
 
 
+Protected Branches
+------------------
+
+Within each repository, the master branch, any branch ending in ``-dev``, and any
+branch ending in ``-release`` should be marked as "protected" on GitHub. There is
```

Comment:
Maybe link the word "protected" to the github page on protected branches?  https://help.github.com/articles/about-protected-branches/